### PR TITLE
Add basic support for arrays

### LINKED ISSUE #20: Implement support for arrays, pointer arithmetic and the * and & operators.
- [x] Implement support for arrays,
- [x] Implement pointer arithmetic 
- [x] Implement * and & operators.

### PR DESCRIPTION
**Pull request by jieffyy**
Add basic support for arrays.

```C
int main() {
    int i;
    int x[5];
    
    for (i = 0; i < 5; i++) {
        x[i] = i;
    }
    
    // returns 4 * 3 = 12
    return x[4] * x[3];
}
```

Arrays are handled in this manner:

Example: `int x[3]`.
This takes up 4 slots.
- The first slot refers to x. It holds a memory address that points to x[0]
- After that, then comes x[0], x[1], ...

Closes #20

### CODE REVIEW

File: src/compiler/typeChecker.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  return ls[0] === '*'
+}
+
+export function throwIfNotPointer(ls: CompileType[]): void {
+  ls.forEach(e => {
+    if (!isPointer(e.typeList)) throw new CompileTimeError()
+  })
+}
+
+export function getBiggerType(t1: CompileType, t2: CompileType): CompileType {
```
Comment (athuyaoo) [inline src/compiler/typeChecker.ts:161]: What does this do?

File: src/compiler/typeChecker.ts [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+  const RANKING = [DataType.CHAR, DataType.SHORT, DataType.INT, DataType.LONG]
+  const assignRanking = (t: DataType) => RANKING.indexOf(t)
+
+  const t1Rank = assignRanking(t1.t)
+  const t2Rank = assignRanking(t2.t)
+
+  return t1Rank >= t2Rank ? t1 : t2
+}
+
+function getLiteralType(node: es.Literal): CompileType {
```
Comment (athuyaoo) [inline src/compiler/typeChecker.ts:171]: What does this code do?

File: src/compiler/compileFunctionDef.ts [hunk 3]
```diff
@@ -5,6 +5,7 @@ import { compileBlkStmt } from './compileStmt'
 import { FunctionCTE, GlobalCTE, VariableInfo } from './compileTimeEnv'
 import { CompileTimeError } from './error'
 import { MICROCODE } from './microcode'
+import { getIdentSize } from './util'
```
Comment (athuyaoo) [inline src/compiler/compileFunctionDef.ts:8]: can we rename this to ful?

**Review by athuyaoo — COMMENTED** [review]
Add some comments for these please and i think we're good

File: src/compiler/compileFunctionDef.ts (hunk 3 above)
Comment (athuyaoo) [reply src/compiler/compileFunctionDef.ts:8]: getIdentificationSize?

**Review by emmaneugene — APPROVED** [review]

File: src/compiler/compileFunctionDef.ts (hunk 3 above)
Comment (jieffyy) [reply src/compiler/compileFunctionDef.ts]: Fixed

File: src/compiler/typeChecker.ts (hunk 2 above)
Comment (jieffyy) [reply src/compiler/typeChecker.ts:192]: Fixed

File: src/compiler/typeChecker.ts (hunk 1 above)
Comment (jieffyy) [reply src/compiler/typeChecker.ts:178]: Fixed

**Pull request merged**